feat: make screenshot test easy to add between steps

alaibe opened the pull request:
Add a decorator to easily add screenshot test.

My suggestion is that after the screenshot have been generated, re-run them and adjust the accuracy if needed (i moved the cursor down to 85% in order to have screenshot test with success)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 90fbb491 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-7201/1/) | 2022-08-31 11:06:27 | ~9 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220831-105725-90fbb4-pr7201.dmg) |

jrainville reviewed (commented):
That's pretty cool.

I just wonder if 85% is too low? There's no way it catches wrong texts. Would it be only to check that we're on the right page? Could we voluntarily mess up the UI a bit to see if it catches big misalignments or something?

I guess what I'm trying to figure out is what is the goal of the screenshot testing for scenarios and does the 85% manage to do that job?

alaibe commented:
@jrainville I updated the decorator with a new parameter and also updated the default view to screenshot. I kept the tolerance at 85%

I made a little change the text and see the error below:
![image](https://user-images.githubusercontent.com/491074/187860789-5390a1f9-18e8-43a8-9544-881475fd3d33.png)

There is many failure, on height and width attribute, on screenshot, and finally on the text attribute.
The checkpoint are not only screenshot test but also attributes test so we should be absolutely fine with 85%

jrainville commented on ui/app/mainui/AppMain.qml:
```diff
@@ -452,7 +454,7 @@ Item {
                 StackLayout {
                     id: appView
-
+                    
```
```suggestion

```

jrainville reviewed (approved):
Very cool!

Khushboo-dev-cpp reviewed (approved):
LGTM :)
